src/Debug.mo

```diff
@@ -1,14 +1,34 @@
-/// Debugging aids
+/// Utility functions for debugging.
+///
+/// Import from the base library to use this module.
+/// ```motoko name=import
+/// import Debug "mo:base/Debug";
+/// ```
 
 import Prim "mo:⛔";
 module {
+  /// Prints `text` to output stream.
+  ///
+  /// NOTE: What this output stream is depends on your execution environment.
+  ///
+  /// ```motoko include=import
+  /// Debug.print "Hello New World!";
+  /// Debug.print(debug_show(4)) // Often used with `debug_show` to convert values to Text
+  /// ```
+  public func print(text : Text) {
+    Prim.debugPrint text;
+  };
 
-  /// `print(t)` emits text `t` to the debug output stream.
-  /// How this stream is stored or displayed depends on the
-  /// execution environment.
-  public let print : Text -> () = Prim.debugPrint;
-
-  /// `trap(t)` traps execution with a user-provided message.
-  public let trap : Text -> None = Prim.trap;
-
+  /// Causes program to trap (error) and ends execution. Prints `errorMessage`
+  /// to output stream.
+  ///
+  /// NOTE: The output is in the replica log. When running on mainnet, it
+  /// is not visible to the developer.
+  ///
```

Comment:
```suggestion
  /// `trap(t)` traps execution with a user-provided diagnostic message.
  /// 
  /// NOTE: the consumer of a future whose execution called `trap(t)` will observe the trap as an `Error` value, thrown at `await`, with code '#canister_error` and message `m`. 
  /// Here `m` is an implementation defined `Text` message derived from `t`).
  ///
  /// NOTE: Other execution environments, that cannot handled Wasm traps, may just propagate the trap and terminate execution, with or without some  descriptive message.
```

 https://m7sm4-2iaaa-aaaab-qabra-cai.raw.ic0.app/?tag=3447622867

Reply:
@kentosugama how is that?

Reply:
Pushed changes